Contents/Resources/Series/Shoko Series Scanner.py

```diff
@@ -111,35 +117,67 @@ def Scan(path, files, mediaList, subdirs, language=None, root=None):
-                showTitle = series_data['name'].encode("utf-8") #no idea why I need to do this.
-                Log.info('show title: %s', showTitle)
-
-                seasonNumber = 0
-                seasonStr = try_get(episode_data, 'season', None)
-                if episode_data['eptype'] == 'Credits': seasonNumber = -1 #season -1 for OP/ED
-                elif episode_data['eptype'] == 'Trailer': seasonNumber = -2 #season -2 for Trailer
-                elif Prefs['SingleSeasonOrdering'] or seasonStr == None:
-                    if episode_data['eptype'] == 'Episode': seasonNumber = 1
-                    elif episode_data['eptype'] == 'Special': seasonNumber = 0
-                else:
-                    seasonNumber = int(seasonStr.split('x')[0])
-                    if seasonNumber <= 0 and episode_data['eptype'] == 'Episode': seasonNumber = 1
-                    elif seasonNumber > 0 and episode_data['eptype'] == 'Special': seasonNumber = 0
-
-                if seasonNumber == 0 and Prefs['IncludeSpecials'] == False: continue
-                if seasonNumber < 0 and Prefs['IncludeOther'] == False: continue #Ignore this by choice.
-
-                if (try_get(series_data, "ismovie", 0) == 1 and seasonNumber >= 1): continue # Ignore movies in preference for Shoko Movie Scanner, but keep specials as Plex sees specials as duplicate
-                Log.info('season number: %s', seasonNumber)
-                episodeNumber = int(episode_data['epnumber'])
-                Log.info('episode number: %s', episodeNumber)
-
-                vid = Media.Episode(showTitle, seasonNumber, episodeNumber)
+                # Get file data using filename
+                # http://127.0.0.1:8111/api/v3/File/PathEndsWith/Clannad/%5Bjoseole99%5D%20Clannad%20-%2001%20(1280x720%20Blu-ray%20H264)%20%5B8E128DF5%5D.mkv
+                filename = os.path.join(os.path.split(os.path.dirname(file))[-1], os.path.basename(file)) # Parent folder + file name
+                file_data = HttpReq('api/v3/File/PathEndsWith/%s' % (urllib.quote(filename)))
+                if len(file_data) == 0: continue # Skip if file data is not found
+
+                # Take the first file. As we are searching with both parent folder and filename, there should be only one result.
+                if len(file_data) > 1:
+                    Log.info('File search has more than 1 result. HOW DID YOU DO IT?')
+                file_data = file_data[0]
+                
+                # Ignore unrecognized files
+                if 'SeriesIDs' not in file_data:
+                    Log.info('Unrecognized file. Skipping!')
+                    continue
+
+                # Get series data
+                series_id = file_data['SeriesIDs'][0]['SeriesID']['ID'] # Taking the first matching anime. Not supporting multi-anime linked files for now. eg. Those two Toradora/One Piece episodes
```

Comment:
Needs to handle errors better

Reply:
Line 132:
SeriesIDs is null for unrecognized files so we are skipping those
If it is not null, then `file_data['SeriesIDs'][0]['SeriesID']['ID']` has to exist.